app/kuma-dp/pkg/dataplane/envoy/remote_bootstrap.go

```diff
@@ -48,10 +48,17 @@ func (b *remoteBootstrap) Generate(url string, cfg kuma_dp.Config) (proto.Messag
 		return nil, errors.Wrap(err, "request to bootstrap server failed")
 	}
 	defer resp.Body.Close()
-	if resp.StatusCode != 200 {
-		if resp.StatusCode == 404 {
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode == http.StatusNotFound {
 			return nil, errors.New("Dataplane entity not found. If you are running on Universal please create a Dataplane entity on kuma-cp before starting kuma-dp. If you are running on Kubernetes, please check the kuma-cp logs to determine why the Dataplane entity could not be created by the automatic sidecar injection.")
 		}
+		if resp.StatusCode == http.StatusUnprocessableEntity {
+			bodyBytes, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, errors.Errorf("Unable to read the response with status code: %d", resp.StatusCode)
+			}
+			return nil, errors.Errorf("Error: %s", string(bodyBytes))
```

Comment:
Is `Error: ` necessary? What will be the output then when I execute kuma-dp and such error will happen?

Reply:
The output is too convoluted in general. I wanted to make it slightly better, but that can mess with other errors.
`kuma-dp` just outputs that it crashed with too many Go internal details (backtrace dumps).